universum/lib/module_arguments.py

```diff
@@ -1,11 +1,20 @@
 # pylint: disable = protected-access
 
+from typing import Dict, Type, TYPE_CHECKING
+
 import argparse
 import os
 import sys
 
+if TYPE_CHECKING:
+    from .gravity import Module  # for static type check
```

Comment:
I don't like cross-dependency between module_arguments and gravity.
Without type annotations, gravity can use any class instance as main_settings, but it is ok to explicitly state it is ModuleNamespace.
However, here we have backward dependency.
Could you, please, find a way to do it without cross dependencies?

Reply:
I removed cross dependency by creating separate interface and inheriting it - similar to what we plan to do with 'needs_structure' and 'needs_output'